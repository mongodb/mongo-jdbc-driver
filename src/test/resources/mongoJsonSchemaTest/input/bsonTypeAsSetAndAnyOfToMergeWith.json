{
  "bsonType": "object",
  "properties": {
    "bar": {
      "bsonType": [
        "int",
        "object",
        "string",
        "bool",
        "null"
      ],
      "anyOf": [
        {
          "bsonType": "null"
        },
        {
          "bsonType": "bool",
          "additionalProperties": true
        },
        {
          "bsonType": "object",
          "properties": {
            "a": { "bsonType": "string" }
          },
          "required": [
            "a"
          ]
        }
      ],
      "properties": {
        "a": {"bsonType": "int"}
      },
      "required": [
        "a"
      ]
    }
  }
}
