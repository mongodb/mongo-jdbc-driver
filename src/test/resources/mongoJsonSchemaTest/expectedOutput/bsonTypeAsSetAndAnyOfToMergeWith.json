{
  "additionalProperties": false,
  "bsonType": "object",
  "properties": {
    "bar": {
      "additionalProperties": false,
      "anyOf": [
        {
          "additionalProperties": false,
          "bsonType": "object",
          "properties": {
            "a": {
              "bsonType": "string"
            }
          },
          "required": [
            "a"
          ]
        },
        {
          "bsonType": "bool"
        },
        {
          "bsonType": "int"
        },
        {
          "additionalProperties": false,
          "bsonType": "object",
          "properties": {
            "a": {
              "bsonType": "int"
            }
          },
          "required": [
            "a"
          ]
        },
        {
          "bsonType": "string"
        },
        {
          "bsonType": "null"
        },
        {
          "bsonType": "bool",
          "additionalProperties": true
        }
      ]
    }
  }
}
